### ISSUE TITLE
Cannot set Metadata when using lib-storage `Upload`.

### ISSUE THREAD

**Issue by peterwmwong** [aws/aws-sdk-js-v3#1891]
**Describe the bug**

`@aws-sdk/lib-storage`'s `Upload` ignores `Metadata` parameter.

**SDK version number**: `@aws-sdk/lib-storage = 3.1.0`

**Is the issue in the browser/Node.js/ReactNative?**
Node.js

**Details of the browser/Node.js/ReactNative version**

```
> node -v
v15.4.0
```

**To Reproduce (observed behavior)**

```ts
import { createReadStream } from 'fs';
import { Upload } from '@aws-sdk/lib-storage';
import { S3 } from '@aws-sdk/client-s3';

(async () => {
    const fileStream = createReadStream(__filename);
    try {
        const upload = new Upload({
            client: new S3({}),
            params: {
                Bucket: 'test-bucket',
                Key: 'test-key',
                Body: fileStream,
                // vvv BUG: Currently ignored! vvv
                Metadata: {
                    testMetaData1: 'one',
                    testMetaData2: 'two',
                },
            },
        });

        await upload.done();
    } catch (e) {
        console.log(e);
    }
})();
```

**Actual behavior**

<img width="773" alt="Screen Shot 2021-01-07 at 10 12 25 AM" src="https://user-images.githubusercontent.com/284734/104029256-5ef1a500-518f-11eb-9072-91fd796ffbcd.png">

**Expected behavior**

Specified `Metadata` is added to the uploaded S3 object: (`x-amx-meta-testmetadata1` and `x-amx-meta-testmetadata2`).

<img width="739" alt="Screen Shot 2021-01-07 at 10 15 03 AM" src="https://user-images.githubusercontent.com/284734/104029284-67e27680-518f-11eb-9366-c5e7024131e5.png">

**Additional context**

I believe the problem is `Uploader.intialize()` is currently NOT considering all the properties provided by `PutObjectCommandInput` (in this case `Metadata`).

https://github.com/aws/aws-sdk-js-v3/blob/d4c302b816e1781f8d04bd479cc4e26e0fe4debc/lib/storage/src/upload/service-clients/Uploader.ts#L50-L57

A monkey patch workaround, we've been using is the following

```diff
import { createReadStream } from 'fs';
import { Upload } from '@aws-sdk/lib-storage';
import { S3 } from '@aws-sdk/client-s3';
import { MultiPartIdentity } from '@aws-sdk/lib-storage/dist/es/upload/service-clients/Uploader';
+import { MultiPartIdentity } from '@aws-sdk/lib-storage/dist/es/upload/service-clients/Uploader';

(async () => {
    const fileStream = createReadStream(__filename);
    try {
        const upload = new Upload({
            client: new S3({}),
            params: {
                Bucket: 'test-bucket',
                Key: 'test-key',
                Body: fileStream,
                // vvv BUG: Currently ignored! vvv
                Metadata: {
                    testMetaData1: 'one',
                    testMetaData2: 'two',
                },
            },
        });

+        // MONKEY PATCH WORKAROUND
+        // =======================
+        //
+        // Override uploader's intialize() to pass (currently omitted) Metadata.
+        //
+        upload.uploader.intialize = async function () {
+            this.destination = (await this._initiateUpload({
+                Bucket: this.destination.Bucket,
+                Key: this.destination.Key,
+                // vvv Currently missing, fixes the bug vvv
+                Metadata: (this as any).configuration.params.Metadata,
+            })) as MultiPartIdentity;
+            return this.destination;
+        };

        await upload.done();
    } catch (e) {
        console.log(e);
    }
})();
```

**Comment by mitchconquer** [comment]
Hi, I am finding the same issue with the ACL property also being ignored.

**Comment by parventures** [comment]
ACL is ignored for me as well.

**Comment by zephod** [comment]
Hi, this is ignoring `params.ContentType` for me, in addition to the properties mentioned above.

**Comment by alchemycs** [comment]
Hey @zephod - I used a slight alteration to the above monkey patch, might be of use for you:

```
      const upload = new Upload({
        client: this.s3Client,
        params: putObjectCommand,
      });
      // START CONTENT-TYPE WORKAROUND https://github.com/aws/aws-sdk-js-v3/issues/1891
      // (this patch is a modification from the monkey patch on the above issue)
      upload.uploader.intialize = async function () {
        this.destination = (await this._initiateUpload({
          Bucket: this.destination.Bucket,
          Key: this.destination.Key,
          // Currently missing, fixes the bug vvv
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          ContentType: (this as any).configuration.params.ContentType,
        })) as MultiPartIdentity;
        return this.destination;
      };
      // END CONTENT-TYPE WORKAROUND
```

**Comment by michaelfarrell76** [comment]
it's honestly a bit baffling that this bug exists and this package is out of beta. when is this expected to be completed? the upgrade process from v2 -> v3 is really painful

**Comment by markb-trustifi** [comment]
Uploader also ignores all encryption-related parameters:
- SSECustomerAlgorithm
- SSECustomerKey
- SSECustomerKeyMD5

**Comment by alexforsyth** [comment]
This should be fixed with this PR: https://github.com/aws/aws-sdk-js-v3/pull/2039

**Comment by andrejunges** [comment]
thanks for the fix @alexforsyth 
one additional question, I got some errors when I upgraded from aws-sdk v2 and I mapped it down to the metadata.
Basically I had one metadata having a boolean value and before sending through the headers, there's some logic (within aws sdk code) to trim the values - so it supports only strings..
https://github.com/aws/aws-sdk-js-v3/blob/main/packages/signature-v4/src/getCanonicalHeaders.ts#L27

is that expected? For now I'm just stringfying all the metadata values.
